Src/VimCore/CommandUtil.fs

```diff
@@ -944,8 +944,13 @@ type internal CommandUtil
 
     /// Get the appropriate register for the CommandData
     member x.GetRegister (commandData : CommandData) =
-        let name = 
+            match commandData.RegisterName with
+            | Some name -> Some (_registerMap.GetRegister name)
+            | None -> None
 
+    /// Get the appropriate register for the CommandData
+    member x.GetRegisterForPut (commandData : CommandData) =

```

Comment:
But we need still two differnt methods for GetRegister. One should  get a concrete register to read value and the other one should get a list of registers to update.